wolfcrypt/src/wc_kyber.c

```diff
@@ -409,16 +405,20 @@ static int kyberkey_encapsulate(KyberKey* key, const byte* msg, byte* coins,
         break;
     }
 
-#ifndef USE_INTEL_SPEEDUP
     if (ret == 0) {
         /* Allocate dynamic memory for all matrices, vectors and polynomials. */
+#ifndef USE_INTEL_SPEEDUP
         at = (sword16*)XMALLOC(((kp + 3) * kp + 3) * KYBER_N * sizeof(sword16),
             key->heap, DYNAMIC_TYPE_TMP_BUFFER);
+#else
+        at = (sword16*)XMALLOC(
+            ((KYBER_MAX_K + 3) * KYBER_MAX_K + 3) * KYBER_N * sizeof(sword16),
```

Comment:
Make them the allocation size the same as above.
KYBER_MAX_K -> kp

Reply:
:heavy_check_mark: